### DISCUSSION TITLE
[@types/node] ts4.8\events.d.ts(106,43) Getting error "A rest parameter must be of an array type." with latest version

### DISCUSSION THREAD

**Discussion by lalit-tavisca** [DefinitelyTyped/DefinitelyTyped#68895] [Issues with a @types package]
Getting following error after using latest version.
```
node_modules/@types/node/ts4.8/events.d.ts:106:43 - error TS2370: A rest parameter must be of an array type.

106                 T[K] extends unknown[] ? (...args: T[K]) => void : never
                                              ~~~~~~~~~~~~~
Found 1 error.
```

Typescript version:- 4.5.4
Library version:- https://www.npmjs.com/package/@types/node/v/20.11.24

It is working as expected with the previous version https://www.npmjs.com/package/@types/node/v/20.10.0

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
Hey, unfortunate outcome of using features supported only in TS4.7 or newer.
See similar: #68889
I assume the same explanation of the problem. You're on the older version of TS here.

**Comment by philzy94** [comment]
node_modules/@types/node/events.d.ts:72:43 - error TS2370: A rest parameter must be of an array type.

72                 T[K] extends unknown[] ? (...args: T[K]) => void : never

This is the error that i have been getting since I update to the latest typescript

**Reply by ww-daniel-mora** [reply]
@philzy94 I was also getting this error. Running `npm outdated` showed my typescript package was 4.5.2 and running `npm upgrade typescript` updated me to 4.9.5 which eliminated the error

**Comment by RicardofVallejo** [comment]
node version in our case

**Comment by Yaman-kassas** [comment]
any updates about this error ?
